File: app/src/main/java/com/asmat/rolando/bakingapp/models/Step.java
package com.asmat.rolando.bakingapp.models;

import android.os.Parcel;
import android.os.Parcelable;

/**
 * Created by rolandoasmat on 8/15/17.
 */

public class Step implements Parcelable {
    private final int id;
    private final String shortDescription;
    private final String description;
    private final String videoURL;
    private final String thumbnailURL;

    public Step(int id,  String shortDescription,  String description,  String videoURL,  String thumbnailURL) {
        this.id = id;
        this.shortDescription = shortDescription;
        this.description = description;
        this.videoURL = videoURL;
        this.thumbnailURL = thumbnailURL;
    }

    public Step(Parcel source) {
        this.id = source.readInt();
        this.shortDescription = source.readString();
        this.description = source.readString();
        this.videoURL = source.readString();
        this.thumbnailURL = source.readString();
    }
    
    public static final Creator CREATOR = (Creator)(new Creator() {

        @Override
        public Step createFromParcel( Parcel source) {
            return new Step(source);
        }

        @Override
        public Step[] newArray(int size) {
            return new Step[size];
        }
    });

    public final int getId() {
        return this.id;
    }

    public final String getDescription() {
        return this.description;
    }

    public final String getShortDescription() {
        return this.shortDescription;
    }

    public final String getVideoURL() {
        return this.videoURL;
    }

    public void writeToParcel(Parcel dest, int flags) {
        if(dest != null) {
            dest.writeInt(this.id);
            dest.writeString(this.shortDescription);
            dest.writeString(this.description);
            dest.writeString(this.videoURL);
            dest.writeString(this.thumbnailURL);
        }
    }

    public int describeContents() {
        return 0;
    }
}

File: app/src/main/java/com/asmat/rolando/rocoton/models/Step.kt
package com.asmat.rolando.rocoton.models

import android.os.Parcel
import android.os.Parcelable

/**
 * Created by rolandoasmat on 7/1/17.
 */

class Step: Parcelable {
    val id: Int
    val shortDescription: String
    val description: String
    val videoURL: String
    val thumbnailURL: String

    constructor(id: Int, shortDescription: String, description: String, videoURL: String, thumbnailURL: String) {
        this.id = id
        this.shortDescription = shortDescription
        this.description = description
        this.videoURL = videoURL
        this.thumbnailURL = thumbnailURL
    }

    constructor(source: Parcel) {
        this.id = source.readInt()
        this.shortDescription = source.readString()
        this.description = source.readString()
        this.videoURL = source.readString()
        this.thumbnailURL = source.readString()
    }


    override fun writeToParcel(dest: Parcel?, flags: Int) {
        if(dest != null) {
            dest.writeInt(id)
            dest.writeString(shortDescription)
            dest.writeString(description)
            dest.writeString(videoURL)
            dest.writeString(thumbnailURL)
        }
    }

    override fun describeContents(): Int {
        return 0
    }

    companion object {
        @JvmField final val CREATOR: Parcelable.Creator<Step> = object : Parcelable.Creator<Step> {
            override fun createFromParcel(source: Parcel): Step{
                return Step(source)
            }

            override fun newArray(size: Int): Array<Step?> {
                return arrayOfNulls(size)
            }
        }
    }

}